phpBB/includes/acp/acp_inactive.php

```diff
@@ -252,6 +241,11 @@ function main($id, $mode)
 					}
 					$db->sql_freeresult($result);
 
+					foreach ($messenger_collection_iterator as $messenger_method)
```

Comment:
I don't think it makes sense to call `save_queue` if the queue was not changed. Therefore I'd recommend placing it before the update query on line 231.

Reply:
But then it wan't change anything in comparison with initial code as `save_queue()` will be called exactly the same times as before, for every used and for every messenger method. Currently it'll be called just as many times as many messenger methods exist.

Reply:
If the query returned an empty result, your code will still call save queue for every messenger method. However, if you place your code in the same location as the previous save_queue call, then it'll will only do that in case the query did not return an empty result. I'm not saying that you should remove the foreach.